src/IIIF/IIIF/Serialisation/Deserialisation/ServiceConverter.cs

```diff
@@ -15,17 +15,37 @@ public class ServiceConverter : ReadOnlyConverter<IService>
         bool hasExistingValue, JsonSerializer serializer)
     {
         var jsonObject = JObject.Load(reader);
+        
+        var atType = jsonObject["@type"];
+
+        var result = ImageServiceTwoResult(serializer, atType, jsonObject);
+        if (result != null) return result;
 
-        var service = IdentifyConcreteType(jsonObject);
+        var service = IdentifyConcreteType(jsonObject, atType);
 
         serializer.Populate(jsonObject.CreateReader(), service);
         return service;
     }
 
-    private static IService? IdentifyConcreteType(JObject jsonObject)
+    private IService? ImageServiceTwoResult(JsonSerializer serializer, JToken? atType, JObject jsonObject)
+    {
+        var typeValue = atType?.Value<string>();
+        if (typeValue is "iiif:Image" or nameof(ImageService2))
+        {
+            // avoids stack overflow exceptions from recursively calling the service converter
+            var thisIndex = serializer.Converters.IndexOf(this);
+            serializer.Converters.RemoveAt(thisIndex);
+            
+            var result = jsonObject.ToObject(typeof(ImageService2), serializer);
+            return result as IService;
+        }
+
+        return null;
+    }
+
+    private static IService? IdentifyConcreteType(JObject jsonObject, JToken? atType)

```

Comment:
Small thing - `ImageServiceTwoResult()` will have called `atType?.Value<string>()` already so should this take a different type?